rclcpp_examples/test/test_single_executable_example.py.in

```diff
@@ -0,0 +1,81 @@
+import io
+from launch import LaunchDescriptor
+from launch.exit_handler import primary_exit_handler
+from launch.launcher import DefaultLauncher
+from launch.output_handler import LineOutput, FileOutput
+
+
+class InMemoryHandler(LineOutput):
+    def __init__(self):
+        super(LineOutput, self).__init__()
+        self.left_over_stdout = b''
+        self.left_over_stderr = b''
+        self.stdout_data = io.BytesIO()
+        self.stderr_data = io.BytesIO()
+
+    def on_stdout_lines(self, lines):
+        self.stdout_data.write(lines)
+
+    def on_stderr_lines(self, lines):
+        self.stderr_data.write(lines)
+
+    def get_description(self):
+        return 'InMemoryHandler'
+
+
+def test_executable():
+    output_handler = InMemoryHandler()
+
+    ld = LaunchDescriptor()
+
+    ld.add_process(
+        cmd=['@RCLCPP_EXAMPLES_EXECUTABLE@', 'test_executable'],
+        name='test_executable',
+        exit_handler=primary_exit_handler,
+        output_handlers=[output_handler],
+    )
+
+    launcher = DefaultLauncher()
+    launcher.add_launch_descriptor(ld)
+    rc = launcher.launch()
+
+    output_lines = []
+
+    expected_output = open('@RCLCPP_EXAMPLES_EXPECTED_OUTPUT@', 'rb').read()
+
+    for line in output_handler.stdout_data.getvalue().splitlines():
+        if line.startswith(b'pid'):
+            continue
+        elif line.startswith(b'================='):
+            continue
+        elif line.startswith(b'Report'):
+            continue
+        elif line.startswith(b'Date'):
+            continue
+        elif line.startswith(b'Description'):
+            continue
+        elif line.strip() == b'':
+            continue
+        elif line.startswith(b'Node'):
+            continue
+        elif line.startswith(b'Process'):
+            continue
+        elif line.startswith(b'Thread'):
+            continue
+        elif line.startswith(b'Internals'):
+            continue
+        elif line.startswith(b'rc'):
+            continue
+        else:
+            output_lines.append(line + b'\r\n')
```

Comment:
Instead of relying on Windows line endings in the fixture I would propose to remove the newlines altogether from the check. Simply split the content using `splitlines()` and perform the comparison on the lists.